File: core/src/main/java/com/ctrip/ferriswheel/core/bean/AxisBand.java
package com.ctrip.ferriswheel.core.bean;

import java.io.Serializable;

public class AxisBand implements Serializable {
    private Interval interval;
    private String label;
    private Color color;

    public AxisBand() {
    }

    /**
     * Construct an AxisBand by copying another AxisBand
     *
     * @param another
     */
    public AxisBand(AxisBand another) {
        this.interval = null;
        this.label = another.getLabel();
        this.color = null;

        if (another.getInterval() != null) {
            this.interval = new Interval(another.getInterval());
        }
        if (another.getColor() != null) {
            this.color = new Color(another.getColor());
        }
    }

    public AxisBand(Interval interval, String label, Color color) {
        this.interval = interval;
        this.label = label;
        this.color = color;
    }

    public Interval getInterval() {
        return interval;
    }

    public void setInterval(Interval interval) {
        this.interval = interval;
    }

    public String getLabel() {
        return label;
    }

    public void setLabel(String label) {
        this.label = label;
    }

    public Color getColor() {
        return color;
    }

    public void setColor(Color color) {
        this.color = color;
    }
}

File: ui/react/src/model/AxisBand.ts
import Interval from "./Interval";
import Color from "./Color";

class AxisBand {
    public interval: Interval;
    public label: string;
    public color: Color;

    public static deserialize(input: any): AxisBand {
        const interval = Interval.deserialize(input.interval);
        const label = input.label;
        const color = typeof input.color !== 'undefined' ?
            Color.deserialize(input.color) : undefined;

        return new AxisBand(interval, label, color);
    }

    constructor(
        interval: Interval = new Interval(),
        label: string = '',
        color: Color = new Color()) {

        this.interval = interval;
        this.label = label;
        this.color = color;
    }
}

export default AxisBand;
